File: FEBRUARY/10-02-2026/Koko Eating Bananas.java
class Solution {
    
    public int kokoEat(int[] arr, int k) {
        
        int l = 1, r = (int)1e6 + 1;
        int res = r;
        
        while(l < r){
            int s = l + (r - l) /2 ;
            if(finished(arr, k, s)) {
                res = s;
                r = s;
            }else l = s + 1;
        }
        return res;
    }
    
    private boolean finished(int[] arr, int k, int s){
        
        int hours = 0;
        for(int a: arr){
            hours += (a % s != 0) ? a / s + 1: a / s;
        }
        return hours <= k;
    }
}

File: FEBRUARY/10-02-2026/Koko Eating Bananas.py
class Solution:
    def kokoEat(self, arr, k):
        low,high=1,max(arr)
        
        while low<=high:
            mid=(low+high)//2
            
            hours=0
            
            for bananas in arr:
                hours+=(bananas+mid-1)//mid
            if hours<=k:
                high=mid-1
            else:
                low=mid+1
        return low